detection_rules/version_lock.py

```diff
@@ -233,7 +233,7 @@ def log_changes(r, route_taken, new_rule_version, *msg):
 
                     # add the min_stack_version to the lock if it's explicitly set
                     if rule.contents.metadata.min_stack_version is not None:
-                        lock_from_file["min_stack_version"] = str(min_stack)
+                        lock_from_file["min_stack_version"] = str(min_stack).rstrip(".0")
```

Comment:
nit: 

```suggestion
                        stripped_version = f"{min_stack.major}{min_stack.minor}"  # mainly because you use similar logic a few times below
                        lock_from_file["min_stack_version"] = stripped_version  # or some other name
```

Reply:
Updated to reflect changes for logging and logic.